Python/import.c

```diff
@@ -2146,7 +2147,8 @@ _imp_find_frozen_impl(PyObject *module, PyObject *name, int withdata)
         Py_RETURN_NONE;
     }
     else if (status == FROZEN_BAD_NAME) {
-        Py_RETURN_NONE;
+        set_frozen_error(status, name);
```

Comment:
This should have a test in Lib/test/test_imp.py.  We should also have one in  one in Lib/test/test_importlib/frozen/test_finder.py, to verify `FrozenImporter.find_spec()` returns None in that case.